GNOME: 3.38 → 40

jtojnar opened the pull request:
https://help.gnome.org/misc/release-notes/40.0/
https://forty.gnome.org/

Fixes: https://github.com/NixOS/nixpkgs/issues/116252

**The changes are reasonably tested but there might be [a regression with fingerprint readers](https://github.com/NixOS/nixpkgs/pull/117081#issuecomment-828232633). If you have one, we would appreciate if you could [help debug it](https://github.com/NixOS/nixpkgs/pull/117081#issuecomment-828356484).**

~~Currently, mostly update script changes and updates produced by it are included. Feel free to test packages, send fixup commits against the branch or [read changelogs](https://github.com/jtojnar/what-changed) and point out possible problems.~~

- [x] gtkmm, glibmm, libsigcxx, libsoup, gtksourceview, libgda, lasem are new parallel-installable series ~~so do not bump them for now.~~

https://hydra.nixos.org/jobset/nixpkgs/gnome#tabs-evaluations

## How to test

1. Clone the nixpkgs repo and checkout the `gnome-40` branch
2. Save the following NixOS configuration file
<details><summary>configuration.nix</summary>


```nix
{ pkgs, config, ... }: {
  environment.systemPackages = with pkgs; [
    gdb
    binutils # readelf
    file
    htop
    less
    git
    rr
    gnomeExtensions.gsconnect
    dfeet
  ];
  environment.enableDebugInfo = true;
  services.xserver = {
    enable = true;
    layout = "cz";
    xkbVariant = "qwerty";

    displayManager.gdm = {
      enable = true;
      debug = true;
    };

    desktopManager.xterm.enable = false;
    desktopManager.gnome3 = {
      enable = true;
      debug = true;
      flashback = {
        enableMetacity = true;
      };
    };
  };

  services.gnome3.core-developer-tools.enable = true;
  services.gnome3.games.enable = true;
  services.openssh.enable = true;
  programs.gpaste.enable = true;
  users.mutableUsers = false;

  users.extraUsers.jtojnar = {
    isNormalUser = true;
    uid = 1000;
    extraGroups = [ "wheel" "networkmanager" ];
    password = "holandaise";
    openssh.authorizedKeys.keys = ["ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDYbOlZydfRRCGCT08wdtPcpfSrgxMc6weDx3NcWrnMpVgxnMs3HozzkaS/hbcZUocn7XbCOyaxEd1O8Fuaw4JXpUBcMetpPXkQC+bZHQ3YsZZyzVgCXFPRF88QQj0nR7YVE1AeAifjk3TCODstTxit868V1639/TVIi5y5fC0/VbYG2Lt4AadNH67bRv8YiO3iTsHQoZPKD1nxA7yANHCuw38bGTHRhsxeVD+72ThbsYSZeA9dBrzACpEdnwyXclaoyIOnKdN224tu4+4ytgH/vH/uoUfL8SmzzIDvwZ4Ba2yHhZHs5iwsVjTvLe7jjE6I1u8qY7X8ofnanfNcsmz/ jtojnar@kaiser"];
  };
  nixpkgs.config.allowAliases = false;

  virtualisation.qemu.options = [ "-device intel-hda -device hda-duplex" ];
  virtualisation.memorySize = 2048;
  virtualisation.diskSize = 8048;
}
```

</details>


3. Run the following command (do not forget to adjust the path to the nixpkgs checkout and the configuration): 
	- On NixOS: `env NIX_PATH=nixpkgs=$HOME/Projects/nixpkgs:nixos-config=path/to/configuration.nix nixos-rebuild build-vm`
	- Outside NixOS: `env NIX_PATH=nixpkgs=$HOME/Projects/nixpkgs:nixos-config=path/to/configuration.nix nix-shell -p '(nixos {}).nixos-rebuild' --run 'nixos-rebuild build-vm'`

Commit:
vala_0_52: init at 0.52.2, make default

Commit:
glib: 2.66.8 -> 2.68.1

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
gnome3: support new versioning scheme in the update script

https://discourse.gnome.org/t/new-gnome-versioning-scheme/4235

jtojnar committed:
maintainers/scripts/update.nix: Add support for filtering pkgs by predicate

Arbitrary predicate is useful for updating decentralised package sets like GNOME.

jtojnar committed:
at-spi2-core: 2.38.0 → 2.40.1

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
amtk: 5.2.0 → 5.3.1

jtojnar committed:
cantarell-fonts: 0.111 → 0.301

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
dconf: 0.38.0 → 0.40.0

jtojnar committed:
gnome3.evince: 3.38.2 → 40.1

jtojnar committed:
evolution: 3.38.4 → 3.40.1

Co-Authored-By: Symphorien Gibol <symphorien+git@xlumurb.eu>

jtojnar committed:
evolution-data-server: 3.38.4 → 3.40.1

jtojnar committed:
evolution-ews: 3.38.3 → 3.40.1

Co-Authored-By: Symphorien Gibol <symphorien+git@xlumurb.eu>

jtojnar committed:
folks: 0.14.0 → 0.15.2

- Had to add libgdata to fix:
	[40/253] Compiling Vala source ../backends/eds/lib/edsf-persona-store.vala ../backends/eds/lib/edsf-persona.vala backends/eds/lib/namespace.vala folks/folks.vapi folks/folks-internal.vapila ../backends/ofono/org-ofono.vala folks/folks.vapi folks/folks-internal.vapi
	FAILED: backends/eds/lib/libfolks-eds.so.26.0.0.p/edsf-persona-store.c backends/eds/lib/libfolks-eds.so.26.0.0.p/edsf-persona.c backends/eds/lib/libfolks-eds.so.26.0.0.p/namespace.c backends/eds/lib/folks/folks-eds.h backends/eds/lib/folks-eds.vapi backends/eds/lib/FolksEds-0.7.gir
	valac -C --vapidir /build/folks-0.15.2/folks -D FOLKS_COMPILATION --pkg libxml-2.0 --pkg libedataserver-1.2 --pkg libebook-contacts-1.2 --pkg libebook-1.2 /build/folks-0.15.2/folks/folks-generics.vapi --pkg gee-0.8 --pkg gio-2.0 --pkg gmodule-2.0 --pkg gobject-2.0 /build/folks-0.15.2/folks/build-conf.vapi --color=always --directory backends/eds/lib/libfolks-eds.so.26.0.0.p --basedir ../backends/eds/lib --library folks-eds --header backends/eds/lib/folks/folks-eds.h --vapi ../folks-eds.vapi --gir ../FolksEds-0.7.gir --includedir folks ../backends/eds/lib/edsf-persona-store.vala ../backends/eds/lib/edsf-persona.vala backends/eds/lib/namespace.vala folks/folks.vapi folks/folks-internal.vapi
	error: Package `libgdata' not found in specified Vala API directories or GObject-Introspection GIR directories
	Compilation failed: 1 error(s), 0 warning(s)
	[41/253] Compiling Vala source ../backends/ofono/ofono-backend-factory.vala ../backends/ofono/ofono-backend.vala ../backends/ofono/ofono-persona-store.vala ../backends/ofono/ofono-persona.vala ../backends/ofono/org-ofono.vala folks/folks.vapi folks/folks-internal.vapi
	FAILED: backends/ofono/ofono.so.p/ofono-backend-factory.c backends/ofono/ofono.so.p/ofono-backend.c backends/ofono/ofono.so.p/ofono-persona-store.c backends/ofono/ofono.so.p/ofono-persona.c backends/ofono/ofono.so.p/org-ofono.c backends/ofono/ofono.h backends/ofono/ofono.vapi
	valac -C --vapidir /build/folks-0.15.2/folks -D FOLKS_COMPILATION --pkg libedataserver-1.2 --pkg libebook-1.2 /build/folks-0.15.2/folks/folks-generics.vapi --pkg gee-0.8 --pkg gio-2.0 --pkg gmodule-2.0 --pkg gobject-2.0 /build/folks-0.15.2/folks/build-conf.vapi --color=always --directory backends/ofono/ofono.so.p --basedir ../backends/ofono --library ofono --header backends/ofono/ofono.h --vapi ../ofono.vapi ../backends/ofono/ofono-backend-factory.vala ../backends/ofono/ofono-backend.vala ../backends/ofono/ofono-persona-store.vala ../backends/ofono/ofono-persona.vala ../backends/ofono/org-ofono.vala folks/folks.vapi folks/folks-internal.vapi
	error: Package `libgdata' not found in specified Vala API directories or GObject-Introspection GIR directories
	Compilation failed: 1 error(s), 0 warning(s)

jtojnar committed:
gjs: 1.66.2 → 1.68.0

jtojnar committed:
glib-networking: 2.66.0 → 2.68.1

jtojnar committed:
gnome-builder: 3.38.2 → 3.40.1

jtojnar committed:
gnome-multi-writer: 3.32.1 → 3.35.90

jtojnar committed:
gnome-online-accounts: 3.38.1 → 3.40.0

jtojnar committed:
gnome-photos: 3.38.0 → 40.0

jtojnar committed:
gnome3.geary: 3.38.1 → 40.0

SuperSandro2000 commented on pkgs/desktops/gnome-3/core/baobab/default.nix:
```diff
@@ -4,13 +4,13 @@
   src = fetchurl {
     url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${name}.tar.xz";
-    sha256 = "0ac3fbl15l836yvgw724q4whbkws9v4b6l2xy6bnp0b0g0a6i104";
+    sha256 = "0000000000000000000000000000000000000000000000000000000000000000";
```
You missed that hash.

jtojnar replied:
For some reason the servers are timing out for me.

SuperSandro2000 commented on pkgs/desktops/gnome-3/apps/gedit/default.nix:
```diff
@@ -23,11 +23,11 @@
   src = fetchurl {
     url = "mirror://gnome/sources/gedit/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "0kc48a399achcz6vlqq0jk0b8ixbrzyv9xb22s5av76m5hyqalq0";
+    sha256 = "0000000000000000000000000000000000000000000000000000000000000000";
```
And you missed that hash.

SuperSandro2000 commented on pkgs/data/documentation/gnome-user-docs/default.nix:
```diff
@@ -9,11 +9,11 @@
   src = fetchurl {
     url = "mirror://gnome/sources/gnome-user-docs/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "1h9lyn80zccmgi6gpymabgrcj7km0sb1axll5z490qnx74xbn37m";
+    sha256 = "0000000000000000000000000000000000000000000000000000000000000000";
```
And you missed that hash.

dotlambda replied:
and the commit message is wrong

jtojnar committed:
gnome3.gnome-klotski: 3.38.1 → 3.38.2

jtojnar committed:
gnome3.gnome-terminal: 3.38.3 → 3.40.1

jtojnar committed:
gnome3.gvfs: 1.46.2 → 1.48.1

jtojnar committed:
gnome3.seahorse: 3.38.0.1 → 40.0

jtojnar committed:
gobject-introspection: 1.66.1 → 1.68.0

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
gthumb: 3.10.2 → 3.11.3

jtojnar committed:
gtk4: 4.0.3 → 4.2.1

Switches to gi-docgen for building docs.

jtojnar committed:
libdazzle: 3.38.0 → 3.40.0

jtojnar committed:
libgee: 0.20.3 → 0.20.4

jtojnar committed:
libgudev: 234 → 236

jtojnar committed:
libpeas: 1.28.0 → 1.30.0

jtojnar committed:
librsvg: 2.50.1 → 2.50.5

jtojnar committed:
libsigcxx: 2.10.1 → 2.10.6

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
mobile-broadband-provider-info: 20190116 → 20201225

Co-Authored-By: Symphorien Gibol <symphorien+git@xlumurb.eu>

jtojnar committed:
python38Packages.pygobject3: 3.38.0 → 3.40.1

jtojnar committed:
sound-juicer: 3.24.0 → 3.38.0

jtojnar committed:
sysprof: 3.38.1 → 3.40.1

jtojnar committed:
tepl: 5.0.1 → 6.00.0

jtojnar committed:
maintainers/scripts/update.nix: Ensure the worktree is clean

When an update script fails, it might still modify the source tree.
These changes would then be committed in the next update attempt.
Let’s make sure the worktree is clean before updating to avoid that.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/what-are-your-goals-for-21-05/11559/17

jtojnar committed:
orca: 3.38.2 → 40.0

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
gnome3.gnome-calendar: 3.38.2 → 40.rc

jtojnar committed:
gnome3.gnome-maps: 3.38.4 → 40.1

jtojnar committed:
gnome3.gnome-boxes: 3.38.2 → 40.1

jtojnar committed:
gnome3.gnome-music: 3.38.2 → 40.0

jtojnar committed:
gnome3.gnome-sound-recorder: 3.38.1 → 40.0

jtojnar committed:
baobab: 3.38.0 → 40.0

jtojnar committed:
epiphany: 3.38.2 → 40.1

jtojnar committed:
gnome3.eog: 3.38.2 → 40.1

jtojnar committed:
gnome3.gnome-initial-setup: 3.38.4 → 40.1

- gnome-tour is now launched by shell https://gitlab.gnome.org/GNOME/gnome-initial-setup/-/issues/120

jtojnar committed:
gnome3.gnome-screenshot: 3.38.0 → 40.0

jtojnar committed:
gnome3.gnome-control-center: 3.38.4 → 40.0

jtojnar committed:
gnome3.nautilus: 3.38.2 → 40.1

jtojnar committed:
gnome3.simple-scan: 3.38.2 → 40.0

jtojnar committed:
gnome-tour: 3.38.0 → 40.0

jtojnar committed:
gnome3.gnome-mines: 3.36.1 → 40.0

jtojnar committed:
gnome3.swell-foop: 3.34.1 → 40.1

jtojnar committed:
gnome3.gnome-tweaks: 3.34.0 → 40.0

jtojnar committed:
gsettings-desktop-schemas: 3.38.0 → 40.0

jtojnar committed:
gnome3.gdm: 3.38.2.1 → 40.0

jtojnar committed:
gnome3.gnome-shell: 3.38.3 → 40.0

jtojnar committed:
gnome-user-docs: 3.38.2 → 40.1

jtojnar committed:
gnome3.gedit: 3.38.1 → 40.1

jtojnar committed:
gnome3.adwaita-icon-theme: 3.38.0 → 40.1.1

jtojnar committed:
gnome3.devhelp: 3.38.1 → 40.0

jtojnar committed:
gnome3.gnome-characters: 3.34.0 → 40.0

jtojnar committed:
gnome3.gnome-backgrounds: 3.38.0 → 40.1

jtojnar committed:
gnome3.gnome-calculator: 3.38.2 → 40.1

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
gnome3.gnome-chess: 3.38.1 → 40.1

jtojnar committed:
gnome3.gnome-clocks: 3.38.0 → 40.0

jtojnar committed:
gnome3.gnome-notes: 3.38.0 → 40.1

jtojnar committed:
gnome3.gnome-contacts: 3.38.1 → 40.0

jtojnar committed:
gnome3.gnome-desktop: 3.38.4 → 40.1

- use libxkbregistry to parse evdev rules files for us https://gitlab.gnome.org/GNOME/gnome-desktop/-/merge_requests/88
- add debug info – I have seen some crashes so let’s make it easier to debug them.

jtojnar committed:
gnome3.gnome-weather: 3.36.1 → 40.0

jtojnar committed:
gnome3.gnome-devel-docs: 3.38.2 → 40.0

jtojnar committed:
gtkmm3: 3.24.3 → 2.24.4

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
python38Packages.pyatspi: 2.38.0 → 2.38.1

jtojnar committed:
tracker: 3.0.3 → 3.1.1

Also reverts #118823

Co-Authored-By: Bastian Köcher <info@kchr.de>
Co-Authored-By: Stefan Frijters <sfrijters@gmail.com>

jtojnar committed:
tracker-miners: 3.0.4 → 3.1.1

jtojnar committed:
vte: 0.62.2 → 0.64.0

jtojnar committed:
gnome3.gnome-disk-utility: 3.38.2 → 40.1

jtojnar committed:
gnome3.gnome-dictionary: 3.26.1 → 40.0

jtojnar committed:
gnome3.gnome-settings-daemon: 3.38.1 → 40.0.1

jtojnar committed:
gnome3.gnome-session: 3.38.0 → 40.1.1

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
gnome3.gnome-software: 3.38.0 → 40.1

jtojnar committed:
gnome3.gnome-shell-extensions: 3.38.2 → 40.0

jtojnar committed:
gnome3.gnome-font-viewer: 3.34.0 → 40.0

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
gnome3.yelp-xsl: 3.38.3 → 40.0

jtojnar committed:
gnome3.yelp: 3.38.3 → 40.0

jtojnar committed:
gnome3.mutter: 3.38.3 → 40.0

jtojnar committed:
gnome3.gnome-system-monitor: 3.38.0 → 40.1

jtojnar committed:
gnome3.gnome-robots: 3.38.0 → 40.0

jtojnar committed:
gnome3.gnome-sudoku: 3.38.0 → 40.1

jtojnar committed:
gnome3.lightsoff: 3.38.0 → 40.0.1

jtojnar committed:
yelp-tools: 3.38.0 → 40.0

Co-Authored-By: Maxine Aubrey <maxeaubrey@gmail.com>

jtojnar committed:
gnome3.tali: 3.38.3 → 40.0

jtojnar committed:
libgweather: 3.36.1 → 40.0

symphorien commented:
it looks like geocode-glib is incompatible with the new version of libsoup
```
Project name: geocode-glib
Project version: 3.26.2
C compiler for the host machine: gcc (gcc 10.2.0 "gcc (GCC) 10.2.0")
C linker for the host machine: gcc ld.bfd 2.35.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Configuring config.h using configuration
meson.build:29: WARNING: Consider using the built-in warning_level option instead of using "-Wall".
Program glib-mkenums found: YES (/nix/store/a9j3wfymvw7bd2wygg9qz7m4xriican5-glib-2.68.0-dev/bin/glib-mkenums)
Found pkg-config: /nix/store/i7sfbrfn7y5dhgd2k19ys0gv2scz64a1-pkg-config-wrapper-0.29.2/bin/pkg-config (0.29.2)
Run-time dependency gio-2.0 found: YES 2.68.0
Run-time dependency json-glib-1.0 found: YES 1.6.2
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency libsoup-2.4 found: NO (tried pkgconfig)

geocode-glib/meson.build:36:0: ERROR: Dependency "libsoup-2.4" not found, tried pkgconfig
```

jtojnar commented:
Yeah, another big bump I forgot about. Reverted for now.

maxeaubrey commented:
> gtkmm, glibmm, libsigcxx, libsoup are new parallel-installable series so do not bump them for now.

Also atkmm and pangomm.

maxeaubrey commented:
`gtksourceview` is a major bump as well, depending on gtk4. We should stick with 4.8 according to upstream.

worldofpeace commented:
I had graham switch the branch for the jobset :+1:

jtojnar committed:
libblockdev: fix build with glib 2.68

jtojnar committed:
gnome3.cheese: fix build

jtojnar committed:
inkscape: Fix build with glib 2.68

jtojnar committed:
gdk-pixbuf: 2.42.2 → 2.42.6

jtojnar committed:
gnome3.gnome-bluetooth: 3.34.3 → 3.34.5

- clean up
- remove intltool dependency https://gitlab.gnome.org/GNOME/gnome-bluetooth/-/commit/e8ad2adfebd6821b459f1d6ba6a8c6f243fcb482

jtojnar committed:
gnome3.libgnome-games-support: 1.8.0 → 1.8.1

jtojnar committed:
umockdev: Fix build with Vala 0.52

jtojnar committed:
umockdev: clean up

jtojnar committed:
zeitgeist: 1.0.2 → 1.0.3

Also fix build with Vala 0.52.

jtojnar committed:
gegl_0_4: 0.4.28 → 0.4.30

Fixes build with GLib 2.68

jtojnar committed:
speechd: Fix build with Glib 2.68

jtojnar committed:
wrapGAppsHook4: init

jtojnar committed:
gjs: fix installed tests

They were already failing on master (with pango 1.47 and gjs 1.66):

	(process:1186): Gjs-CRITICAL **: 12:36:09.500: JS ERROR: Error: Requiring Gtk, version 3.0: Typelib file for namespace 'HarfBuzz', version '0.0' not found
machine # @/nix/store/mca5jqi26f3h3s78p54bp59x4klyrch5-gjs-1.66.2-installedTests/libexec/installed-tests/gjs/js/testLegacyGObject.js:9:13

HarfBuzz is likely pulled in through Pango. Possibly introduced in https://github.com/NixOS/nixpkgs/pull/93799. But then why did not I notice that in GNOME 3.38 bump?

jtojnar committed:
gnome3.dconf-editor: 3.38.2 → 3.38.3

jtojnar committed:
gnome3.gnome-nibbles: fix build with latest Vala

jtojnar committed:
gnome3.gnome-taquin: fix build with latest Vala

jtojnar committed:
gnome3.iagno: fix build with latest Vala

jtojnar commented:
Just finished reviewing the changelogs. Looks mostly good, added the issues I noticed into the project: https://github.com/NixOS/nixpkgs/projects/40

jtojnar committed:
gnome3.gnome-applets: 3.38.0 → 3.40.0

jtojnar committed:
gnome3.gnome-flashback: 3.38.0 → 3.40.0

jtojnar committed:
gnome3.gnome-panel: 3.38.0 → 3.40.0

jtojnar committed:
gnome3.metacity: 3.38.0 → 3.40.0

jtojnar committed:
gnome3.gnome-books: 3.34.0 → 40.0

jtojnar committed:
gnome3.gnome-keyring: 3.36.0 → 40.0

jtojnar committed:
gcr: 3.38.1 → 3.40.0

SFrijters committed:
pythonPackages.tappy: init at 3.0

jtojnar committed:
grilo-plugins: 0.3.12 → 0.3.13

SuperSandro2000 commented:
It got brought to my attention that many extensions will break with gnome 40 like https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome-3/extensions/dash-to-panel/default.nix. How will that be handled by the update?

jtojnar commented:
Extensions that are incompatible with GNOME 40 will be automatically disabled at runtime.

jtojnar committed:
gnome3.cheese: clean up

Abdillah committed:
gpaste: 3.38.5 → 3.40.1

jtojnar committed:
pango: 1.48.3 → 1.48.4

jtojnar committed:
libgweather: clean up

jtojnar commented:
Updated few more packages and rebased to resolve the conflicts.

jtojnar committed:
rl-2105: Mention GNOME 40

jtojnar committed:
pitivi: 2020.09.2 → 2021.01

jtojnar committed:
gtranslator: 3.38.0 → 40.0

jtojnar committed:
libpst: 0.6.75 → 0.6.76

jtojnar committed:
libhandy: 1.2.0 → 1.2.2

jtojnar committed:
libhandy: build glade catalog separately

So that libhandy itself does not have webkitgtk in build time closure through glade.

jtojnar commented:
libhandy no longer depends on webkitgtk through glade.

Confirmed it works using `env GLADE_CATALOG_SEARCH_PATH=$(nix-build -A libhandy.glade --no-out-link)/share/glade/catalogs GLADE_MODULE_SEARCH_PATH=$(nix-build -A libhandy.glade --no-out-link)/lib/glade/modules $(nix-build -A glade --no-out-link)/bin/glade`.

jtojnar committed:
appstream: 0.14.0 -> 0.14.3

Fixes build with recent glib. Also clean up.

jtojnar committed:
libunity: unstable-2019-03-19 → unstable-2021-02-01

Fix build with latest Vala.

jtojnar committed:
pantheon.appcenter: Fix build

Use upstream patch.

To make it apply, we need to update the src hash, which r-ryantm did not do during the version bump: https://github.com/NixOS/nixpkgs/pull/101311

jtojnar committed:
pidgin-sipe: Fix build with latest GLib

jtojnar committed:
gnome3.sushi: Remove clutter dependency

Not used since 3.33.0.

jtojnar committed:
libvirt-glib: 3.0.0 → 4.0.0

jtojnar committed:
deja-dup: 42.6 → 42.7

https://gitlab.gnome.org/World/deja-dup/-/tags/42.7

jtojnar committed:
gnome3.gnome-getting-started-docs: drop

It has been retired

https://gitlab.gnome.org/GNOME/gnome-build-meta/-/issues/353

jtojnar committed:
vala_0_48: 0.48.14 → 0.48.17

jtojnar committed:
nixos/gnome3: Enable power-profiles-daemon

GNOME 40 added support for it in Control Center.

jtojnar committed:
gnomeExtensions.gsconnect: 44 → 46

jtojnar committed:
nixos/gnome3: Install GNOME Tour

It will be run after startup.

jtojnar committed:
gtk-vnc: 1.0.0 → 1.2.0

jtojnar committed:
vala-lint: 2020-08-18 → 2021-02-17

jtojnar committed:
font-manager: 0.8.5-1 → 0.8.6

Fixes a crash with GNOME 40.

https://github.com/FontManager/font-manager/releases/tag/0.8.6

Had to backport patches for the settings to actually work.

https://github.com/FontManager/font-manager/issues/215

jtojnar committed:
xdg-desktop-portal-gtk: Clarify gsettings schema deps

org.gnome.settings-daemon.peripherals.mouse and font-related settings of org.gnome.settings-daemon.plugins.xsettings were moved to gsettings-desktop-schemas in GNOME 40, making the remaining whitelisted schemas from g-s-d basically useless.

https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/issues/31

jtojnar committed:
pantheon.elementary-settings-daemon: backport patches for latest libgweather API

jtojnar committed:
gitg: disable tests

They started faiing with the same issue as before they were enabled in 2019.
I tried running them in xvfb and dbus session (see e.g. libdazzle expression) but without success.
I wonder how it could have worked for so long.ů

I also cleaned up the expression and corrected the license while at it.

jtojnar commented:
Rebased, hopefully for the last time. I tested it somewhat and I believe @maxeaubrey have been also using it so I would consider it good to go once hydra chews through the jobset.

maxeaubrey commented:
Can confirm, I've been running it as a daily driver on my personal laptop for the last week without much issue - I had one non-reproducible instance of logging in taking a very long time, but without anything in the logs, so... :woman_shrugging: 

We might want to revert librsvg and bump it to 2.50.4 instead? Right now it's on unstable, and afaict 2.50.x is still used for the GNOME 40 release, based on build-meta.

jtojnar committed:
librsvg: register installed tests

jtojnar commented:
Thanks for pointing out the librsvg issue. I waited for the stable version to materialize but it never did. Guess it is more rebuilds 🤷‍♀️

Checked all the packages whose version matches `\.\d*[13579]\.|alpha|beta|rc|\.9` to see whether it should be safe to use them:

- tracker – stable
- tracker-miners – stable
- grilo-plugins – stable
- amtk – stable
- folks – unstable, important clean ups
- gnome-multi-writer – rc, only a bugfix
- gnome-online-accounts – rc, only few bugfixes
- gnome3.geary – rc, looks stable enough, important fixes
- vte – rc
- gthumb – stable?
- gnome-calendar – rc, as stable as it gets
- adwaita-icon-theme – rc
- gnome3.file-roller – alpha, fixes cve

maxeaubrey commented:
`vte` also has 0.64.0. I'll open a PR fixing that and librsvg in a bit.

jtojnar commented:
> `vte` also has 0.64.0.

gnome-calendar too but they did not push tarballs.

> I'll open a PR fixing that and librsvg in a bit.

 Already done with the rebase.

Abdillah commented:
In my case, there still some error,

- the gnome-shell missing EGL header (`mesa` dependency fixed this)
- metacity missing `xorg.libXdestroy`
- gnome-flashback missing `xorg.libXxf86vm`

And I'm wondering on how to make the animation smoother in Qemu? It seems increasing the `memsize` doesn't help.

jtojnar committed:
gnome3.gnome-session-ctl: 3.38.0 → 40.0

jtojnar commented:
Since staging-next has been merged, we can target this to master now. cc @vcunat @FRidh

jonringer commented:
@GrahamcOfBorg eval

jonringer commented:
if the rebuild is correct, this should not be targeting master.

jtojnar commented:
@jonringer It has already been built by Hydra: https://hydra.nixos.org/jobset/nixpkgs/gnome#tabs-evaluations

jonringer commented:
Ah, wasn't aware. Sounds good then :)

jtojnar committed:
gnome-latex: fix build with tepl 6

Applying a patch since the app has been archived and will receive no further releases.

Had to rebuild the build scripts with autoreconfHook since the patch modifies configure.ac.

Also cleaned up the expression a bit.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/what-are-your-goals-for-21-05/11559/19

tenaf0 commented:
I get an occasional error with programs that try to open the gtk file chooser dialog. It happens with programs built on non-gnome-40 branch like firefox, chromium and krita and it is fixed by building it on gnome-40. So it's not a show-stopper, just wanted to mention it here.

Here is the error message in case of krita:
```
(.krita-wrapped:19377): Gtk-WARNING **: 12:19:55.482: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /nix/store/c20nvqh5cbzwbi5jcmz8agd6xw7glsji-adwaita-icon-theme-40.0/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /nix/store/c20nvqh5cbzwbi5jcmz8agd6xw7glsji-adwaita-icon-theme-40.0/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Aborted (core dumped)
```

And in case of chromium:
```
Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /nix/store/24xvj3dmqpxv1ypa7wsj7iy937s6x5xh-adwaita-icon-theme-3.38.0/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /nix/store/24xvj3dmqpxv1ypa7wsj7iy937s6x5xh-adwaita-icon-theme-3.38.0/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Aborted (core dumped)
```

dasj19 commented:
I will try this in a VM... what is left to be tested? ... I could try my usual gnome setup that I have in configuration.nix

jtojnar commented:
Me and @ maxeaubrey have been running it on our production systems without a hitch for a while now so I would consider this ready to merge. Of course more testing is always welcome.

There are still some [failing builds](https://hydra.nixos.org/eval/1665478#tabs-still-fail), a few due to the new GLib but those are mostly leaf packages so I would not bother much with them.

dasj19 commented:
So far I found an error in file-roller.
Steps to reproduce:
1. Download the zip file of gnome-40 branch from github: https://github.com/NixOS/nixpkgs/archive/refs/heads/gnome-40.zip
2. Open the archive with Archive Manager (file-roller)
3. Notice the failure to unzip: `An error occured while extracting files Fatal error`
Workaround: Right click on the archive and then extract.

jtojnar commented:
I am pretty sure I have seen that in 3.38 too.

vcunat commented:
I see many darwin regressions due to librsvg not building: https://hydra.nixos.org/build/141650131

dasj19 commented:
@jtojnar I tried a couple of things with file-roller:
1. Running `xnn5k0i84b7003q3l5qa18fgvh1zrrll-file-roller-3.38.0` from the unstable branch works well with that particular .zip file
2. If I install `unzip` it works well with the file-roller from gnome-40 branch
3. When running from console the following appears: `(file-roller:6709): GLib-GIO-CRITICAL **: 08:17:55.625: g_output_stream_write_all: assertion 'buffer != NULL' failed`

All of theese point findings point to a regression

jtojnar commented:
I dropped the file-roller 3.39.1 commit to avoid conflict with 3.38.1 from
master. Could you please check if that version is okay?

On Tue, 27 Apr 2021, 08:35 dasj19, ***@***.***> wrote:

> @jtojnar <https://github.com/jtojnar> I tried a couple of things with
> file-roller:
>
>    1. Running xnn5k0i84b7003q3l5qa18fgvh1zrrll-file-roller-3.38.0 from
>    the unstable branch works well with that particular .zip file
>    2. If I install unzip it works well with the file-roller from gnome-40
>    branch
>    3. When running from console the following appears: (file-roller:6709):
>    GLib-GIO-CRITICAL **: 08:17:55.625: g_output_stream_write_all: assertion
>    'buffer != NULL' failed
>
> All of theese point findings point to a regression
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/117081#issuecomment-827351637>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFMEYYA652SOIZTBGJYOLDTKZLKDANCNFSM4ZQ7IUHA>
> .
>

jtojnar commented on pkgs/development/libraries/librsvg/default.nix:
```diff
@@ -1,32 +1,58 @@
+    vala
+    gobject-introspection
+  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+    ApplicationServices
```
> I see many darwin regressions due to librsvg not building: https://hydra.nixos.org/build/141650131

 Maybe adding Foundation here would fix that?

vcunat replied:
Bad luck, another issue: https://hydra.nixos.org/build/142180721/nixlog/1/tail

It's hard to do this without some access to a darwin machine

SuperSandro2000 commented on pkgs/development/libraries/librsvg/default.nix:
```diff
@@ -1,32 +1,58 @@
+    cargo
+    vala
+    gobject-introspection
+  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
```
darwin shouldn't be used as an input.

maralorn replied:
Can you explain what harm it does?

hmenke replied:
No harm per se, but the same reason you don't use `pkgs` as an input. Instead use `ApplicationServices` as an input directly.

SuperSandro2000 replied:
> Can you explain what harm it does?

It is harder to override it.

jtojnar replied:
Done. Could you please check if adding the Foundation was sufficient?

dasj19 commented:
> I dropped the file-roller 3.39.1 commit to avoid conflict with 3.38.1 from master. Could you please check if that version is okay?
Unfortunately it didn't help. It helps having `unzip` or `p7zip` installed on the system.
It seems that the issue is much deeper in gnome libraries... and you reported it yourself here: https://gitlab.gnome.org/GNOME/file-roller/-/issues/40 ... I'm not knowledgeable enough but it might have roots in glib https://gitlab.gnome.org/GNOME/glib/-/issues/249
With that being said I do not know how to fix this issue.

dasj19 commented:
Another minor setback is the configuration over here: https://github.com/NixOS/nixpkgs/pull/67389#issuecomment-524565602 no longer works. The xdg folders do not get updated when changing it on a live system.

jtojnar commented:
As a stop gap, we could add unzip to PATH of file roller?

dasj19 commented:
> As a stop gap, we could add unzip to PATH of file roller?

Yes, that seems to help. While we are at it we should also add `cpio` to `buildInputs` because it says in the build output that it misses it.

jtojnar commented:
Feel free to push that here while I try to set up https://github.com/fastai/fastmac.

jtojnar commented:
Okay, now it fails with less clear error:

```
    Finished release [optimized + debuginfo] target(s) in 5m 01s
/nix/store/14s1hh53dr8xkvxq3vs5rm9i6nnk4kar-cctools-port-949.0.1/bin/ranlib: file: .libs/librsvg_c_api.a(_rsvg_dummy.o) has no symbols
ranlib: file: .libs/librsvg_c_api.a(_rsvg_dummy.o) has no symbols
warning: ranlib: warning for library: .libs/librsvg_c_api.a the table of contents is empty (no object file members in the library define global symbols)
  CCLD     librsvg-2.la
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
  CCLD     rsvg-convert
  GISCAN   Rsvg-2.0.gir
  GICOMP   Rsvg-2.0.gir
 VAPIGEN librsvg-2.0.vapi
GdkPixbuf-2.0.gir:6346.55-6346.55: error: The symbol `GModule' could not be found
make[2]: *** [/nix/store/zc1llkjhjwqz9nv9a50a8hyjpqdpw21a-vala-0.52.2/share/vala/Makefile.vapigen:61: librsvg-2.0.vapi] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-librsvg-2.50.4.drv-0/librsvg-2.50.4'
make[1]: *** [Makefile:1276: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-librsvg-2.50.4.drv-0/librsvg-2.50.4'
make: *** [Makefile:827: all] Error 2
builder for '/nix/store/6ph2scihzblq2pd0h86aicjvjzd3nrnx-librsvg-2.50.4.drv' failed with exit code 2
```

jtojnar commented:
> Another minor setback is the configuration over here: [#67389 (comment)](https://github.com/NixOS/nixpkgs/pull/67389#issuecomment-524565602) no longer works. The xdg folders do not get updated when changing it on a live system.

I do not think it ever worked that way. It probably only has effect on login (IIRC some GNOME component is asking user to confirm that and then stores the values to `~/.local/share` overriding the config.)

jtojnar commented:
@vcunat I disabled building the VAPI for now but not sure if something does not depend on that.

jtojnar committed:
librest: enable on Darwin

Not sure if it will work since the certificates file is not present on MacOS AFAICT but it is needed for GTK 4 so 🤷‍♀️.

Also clean up the expression and correct license.

vcunat commented:
Now [it builds](https://hydra.nixos.org/eval/1665972#tabs-now-succeed), so it's certainly better without vapi than not at all.

jtojnar committed:
gst_all_1.gst-plugins-bad: fix build on Darwin

jtojnar committed:
webkitgtk: Try to fix build on Darwin

- Libmanette and systemd features need to be disabled since they depend on systemd.
- FastMalloc for some reason does not include <malloc/malloc.h> so we added that.
- <CommonCrypto/CommonRandom.h> depends on a type from <CommonCrypto/CommonCryptor.h> but does not include it for some reason, resulting in the following error:

	In file included from ../Source/WTF/wtf/RandomDevice.cpp:44:
	/nix/store/wd30p2pq4lci8fk9fqldkgk1hgmwpapj-Libsystem-1238.60.2/include/CommonCrypto/CommonRandom.h:35:9: error: unknown type name 'CCCryptorStatus'
	typedef CCCryptorStatus CCRNGStatus;

- The GL-related flags were renamed and removed (in previous releases as well) so we switched to the new ones so that GL is still disabled.
- Unrelatedly, `with` statement at the top of the expression is abomination so we narrowed its scope to just meta.

jtojnar committed:
poppler_gi: fix build on Darwin

jtojnar committed:
calls: Fix build

jtojnar committed:
elementary-planner: fix build

jtojnar committed:
pantheon.elementary-music: Fix build with latest Vala

jtojnar committed:
pantheon.elementary-calendar: fix build

jtojnar committed:
pantheon.switchboard-plug-printers: Fix build with latest Vala

jtojnar committed:
pantheon.switchboard-plug-security-privacy: Fix build with latest Vala

jtojnar committed:
pantheon.elementary-photos: Fix build with latest Vala

jtojnar committed:
pantheon.elementary-code: Fix build with latest Vala

jtojnar committed:
pantheon.wingpanel-indicator-datetime: Fix build

jtojnar committed:
gnucash: 4.4 → 4.5

Needed a patch to fix build with GLib 2.68 and it did not apply to 4.4 so I also bumped the version 🤷‍♀️

Changes look fine: https://github.com/Gnucash/gnucash/releases/tag/4.5

jtojnar commented:
Fixed Pantheon DE and some other packages and it looks like the only remaining potential blocker is `libsForQt5.qtwebkit`. cc @periklis @abbradar @NixOS/qt-kde

jtojnar committed:
gimp: 2.10.22 → 2.10.24

https://www.gimp.org/news/2021/03/29/gimp-2-10-24-released/

vcunat commented:
Speaking of blockers, `nixpkgs-unstable` channel would still be blocked due to `imagemagick.x86_64-darwin`: https://hydra.nixos.org/build/142193914#tabs-summary
(but the webkit issue seems more serious _to me_; I'm not a mac fan really)

LEXUGE commented:
Might be an unrelated issue, but I tested out the GNOME 40 with fprintd enabled. The building worked fine, however, the login process was unsuccessful both in GDM and lock screen. If I type a few words in the password box, the box "shaked" with prompt like "Fingerprind didn't match, please try again".

jtojnar commented:
qtwebkit is the old unmaintained project qtwebview is the new one IIRC so maybe it is not that bad.

jtojnar committed:
imagemagick: Try to fix Darwin build

jtojnar commented:
Yeah, adding Foundation seems to have fixed imagemagick.

This one looks gnarlier: https://hydra.nixos.org/build/142193912/nixlog/1

jtojnar committed:
Fix tarball build

replaceDependency is based on IFD so it cannot be used in packages.
Thankfully what we are doing with libhandy can be easily achieved with sed.

jtojnar commented:
> If I type a few words in the password box, the box "shaked" with prompt like "Fingerprind didn't match, please try again".

Do you mean that neither fingerprint nor regular password worked?

LEXUGE commented:
> > If I type a few words in the password box, the box "shaked" with prompt like "Fingerprind didn't match, please try again".
> 
> @LEXUGE Do you mean that neither fingerprint nor regular password worked?

Yes, in other words, I have to rollback to login in.

jtojnar commented:
That is quite serious. Could you please try enabling debugging with

```nix
services.xserver.displayManager.gdm.debug = true;
services.xserver.desktopManager.gnome3.debug = true;
```

and check the systemd journal (e.g. in TTY or over SSH)?

LEXUGE commented:
@jtojnar Would you like to point out journals of what units/services should I be looking at? I will test it out as soon as I am available.

jtojnar commented:
IIRC it might not be associated with any unit. I would look for anything with gdm/gnome-shell/gnome-session/pam/gcr/fprintd.

dasj19 committed:
file-roller: fixed fatal error occuring when extracting zip files

dasj19 commented:
I don't know if this is relevant... but if I setup a keyboard layout via Gnome Settings it does not apply to the gdm login screen which still uses en_US. There is no locale-related settings in my configuration.nix

davidak commented:
>the gdm login screen which still uses en_US. There is no locale-related settings in my configuration.nix

@dasj19  `en_US` is the default. you probably have to use `i18n.defaultLocale = "en_US.UTF-8";` to change it globally

davidak commented:
>GNOME 40

>`desktopManager.gnome3`

i argued in the iso name discussion that the 3 is an implementation detail the end user don't have to see. they just want gnome and should get gnome. now we have the situation that the user enables `gnome3` but gets GNOME 40. that is inconsistent and therefore a quality issue

when it's not planned to offer multiple gnome versions, i suggest to just call it `desktopManager.gnome` now instead of changing it to `desktopManager.gnome40`

jtojnar commented:
Yeah, I wanted to rename the attributes before even starting to work on the update but then I forgot about this. Will need to create ton of aliases. We should probably do that now while we are waiting for someone to double check the fingerprint issue.

LEXUGE commented:
@jtojnar I tested that out and that was even more serious: after several crashes (restart) in GDM screen, I logged in without my finger or password
Log below:
[log.txt](https://github.com/NixOS/nixpkgs/files/6405387/log.txt)

blitz commented:
I have tested this on my desktop with AMD graphics. I've seen no issues so far. (I don't have a fingerprint reader, so this is no data point for that issue.)

Thanks for all the work that went into this update!

jtojnar committed:
gnomeExtensions.hot-edge: init at 4

jtojnar commented:
@LEXUGE the logs do not say much to me. Could you please open a confidential issue at https://gitlab.gnome.org/GNOME/gdm/-/issues and describe the reproduction steps there? Hopefully, upstream will be able to help us.

jtojnar committed:
gnome3.file-roller: 3.38.1 → 3.40.0

jtojnar committed:
gnome3.gnome-autoar: 0.3.1 → 0.3.2

jtojnar committed:
gnome3.gnome-shell: Switch to master

It contains several critical fixes.

jonringer commented:
since gnome is our primary DM for the graphical iso, I'm going to add this as a release blocker

jtojnar committed:
gnome3.gnome-shell: Pick patch from Fedora

Hoping it will fix the login with fprintd enabled.

jtojnar commented:
@LEXUGE I tried the Fedora workaround patch on my sister’s laptop, and while it still complains, at least it does not seem to be clearing the password field as described on https://gitlab.gnome.org/GNOME/gdm/-/issues/691 any more (I did reproduce it with unpatched 40.0). I managed to log in with fingerprint as well.

Could you verify if it fixes the issue you had?

LEXUGE commented:
Sure, I will check it out today as soon as possible.

jonringer commented:
I used the github UI to resolve the merge conflict. If it causes issues, feel to revert / drop it.

Just wanted to start a nixpkgs-review

jtojnar committed:
libsigcxx30: init at 3.0.6

jtojnar committed:
glibmm_2_68: init at 2.68.0

jtojnar committed:
cairomm_1_16: init at 1.16.0

jtojnar committed:
pangomm_2_48: init at 2.48.0

jtojnar committed:
gtkmm4: init at 4.0.1

jtojnar committed:
gjs: 1.68.0 → 1.68.1

jtojnar committed:
gnome-builder: 3.40.1 → 3.40.2

jtojnar committed:
vte: 0.64.0 → 0.64.1

jtojnar committed:
gtksourceview5: init at 5.0.0

jtojnar committed:
gnome3.gpaste: 3.40.1 → 3.40.2

http://www.imagination-land.org/posts/2021-04-30-gpaste-3.40.2-released.html

jtojnar committed:
gtksourceview4: clean up

Make consistent with gtksourceview5.

fabianhjr commented:
Rebased with master and tried it on my system successfully. No issues so far; didn't even changed my config.

Thanks for the merge request. :+1:

fabianhjr reviewed (approved):
Rebased with master and tried it on my system successfully. No issues so far; didn't even changed my config.

Thanks for the merge request. :+1:

jtojnar commented:
@LEXUGE oh, sorry. GitHub does not notify of edits, usually.

Thanks for confirming it fixed the password issue. Since it is now possible to log in, I would no longer consider it a blocker and try to debug after this was merged.

For what it is worth, I did manage to log in using fingerprint so it would be nice if you could get a fresh log (with the patches applied) so that we could compare them.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/gnome-3-38-on-latest-release/13469/3
